tests/ui/needless_return.fixed

```diff
@@ -81,7 +81,7 @@ fn test_void_if_fun(b: bool) {
 fn test_void_match(x: u32) {
```

Comment:
I feel like none of these tests cover the actual case of the issue. Could you add this test:

```rs
fn test_match_as_stmt() {
    let x = 9;
    match x {
        1 => 2,
        2 => return,
        _ => 0,
    };
    // The semicolon makes it a statement, and allows the branches to be non-unit
}
```

Reply:
Oops messed up a commit somewhere while git worktree shenanigans, updated the PR with your test, thanks for this! 🙇🏻